core/sql/sqlcomp/nadefaults.cpp

```diff
@@ -1958,6 +1958,7 @@ SDDkwd__(EXE_DIAGNOSTIC_EVENTS,		"OFF"),
 
   DDkwd__(HIVE_DEFAULT_CHARSET,            (char *)SQLCHARSETSTRING_UTF8),
   DD_____(HIVE_DEFAULT_SCHEMA,                  "HIVE"),
+  DD_____(HIVE_FILE_CHARSET,            (char *)SQLCHARSETSTRING_UTF8),
```

Comment:
yes, Hans is correct. The current logic cannot work well if in some cases, HIVE_DEFAULT_CHARSET is not UTF8. It is better to change HIVE_FILE_CHARSET as empty and change the checking logic in the synthType.cpp logic accordingly.